### ISSUE TITLE
IRoutingContext is missing some kind of intent

### ISSUE THREAD

**Issue by johnsimons** [Particular/NServiceBus#3717]
As part of addressing https://github.com/Particular/NServiceBus.SqlServer/issues/202 we found that we may need some kind of "intent" property in `IRoutingContext` to tell us what is the reason for this routing.

An [attempt](https://github.com/Particular/NServiceBus.SqlServer/pull/220) by @SimonCropp at fixing this issue  has highlighted this need.

So should we have a `IRoutingContext.RoutingIntent` enum ?

Something like:
```c#
RoutingIntent.Audit
RoutingIntent.Error
RoutingIntent.Callback
RoutingIntent.Forwarding
RoutingIntent.Endpoint
...
```

**Comment by andreasohlund** [comment]
I like it!

**Comment by danielmarbach** [comment]
Having an intent is good. The question is whether using an enumeration could be too restrictive. I know magic strings have their problems too. But maybe using strings for the routing intent would be more flexible or types would work as well

**Comment by andreasohlund** [comment]
Thinking about this a bit more I wonder if this is such a good idea. Why would the transport want to have any smarts depending on how the message is to be routed?

Also keep in mind that we do plan to push most of this into the transports:

* Moving to error should be a transport concern
* Same for auditing and forwarding since that would allow transports to make use of native mechanism to make this more effective.

In short: Not sure we need this in the interim. I'd propose that we wait an see if the need comes up again before taking any actions?

**Comment by timbussmann** [comment]
@danielmarbach and I both agree with @andreasohlund. The idea is interesting but there is no directly visible value in doing so. So let's close this and tackle this when we encounter a situation which needs this.
